### PR TITLE
Assertion condition

### PR DESCRIPTION
**Pull request by mendi160**

### CODE REVIEW

File: athenz/assertion.go [hunk 1]
```diff
@@ -90,11 +171,88 @@ func resourceAssertionSchema() *schema.Schema {
...
 				},
+				"id": {
+					Type:     schema.TypeInt,
+					Computed: true,
+				},
+				"condition": {
+					Type:     schema.TypeSet,
+					MaxItems: 2, // each assertion represent acl policy. Since in a given service,
+					// you can make the acl policy enforced on some hosts and not on others,
+					// therefor, to apply more than 2 conditions per assertion doesn't make sense
```
Comment (dvirguttman) [inline athenz/assertion.go:182]: `therefore`
what happens if there are more than 2 items?

File: athenz/assertion.go [hunk 2]
```diff
@@ -90,11 +171,88 @@ func resourceAssertionSchema() *schema.Schema {
...
+								Type:     schema.TypeSet,
+								Required: true,
+								MaxItems: 1,
+								Elem:     resourceConditionSchema(validateDiagFuncForScope),
+							},
+							"scopeall": {
+								Type:     schema.TypeSet,
+								Required: true,
+								MaxItems: 1,
+								Elem:     resourceConditionSchema(validateDiagFuncForScope),
```
Comment (dvirguttman) [inline athenz/assertion.go:221]: maybe I am missing something - but why these fields are set with only 1 item?

File: athenz/assertion.go [hunk 3]
```diff
@@ -90,11 +171,88 @@ func resourceAssertionSchema() *schema.Schema {
...
+					},
+				},
 			},
 		},
 	}
 }
 
+func expandAssertionConditions(configured []interface{}) *zms.AssertionConditions {
+	conditionsList := make([]*zms.AssertionCondition, 0, len(configured))
+	keys := []string{Instances, EnforcementState, ScopeONPREM, ScopeAWS, ScopeALL}
```
Comment (dvirguttman) [inline athenz/assertion.go:233]: what are these keys?

Comment (dvirguttman) [reply athenz/assertion.go:233]: Now I see its definition: `assertion conditions data keys`
But my question is- the order is matter ?

File: athenz/resource_policy_test.go [hunk 4]
```diff
@@ -19,19 +19,19 @@ func TestAccGroupPolicyBasic(t *testing.T) {
 		log.Print("TF_ACC must be set for acceptance tests")
 		return
 	}
```
Comment (dvirguttman) [inline athenz/resource_policy_test.go:21]: please make sure to not leave it like that

File: msd_module/msd_policy/main.tf [hunk 5]
```diff
@@ -0,0 +1,33 @@
+module "inbound" {
```
Comment (dvirguttman) [inline msd_module/msd_policy/main.tf:1]: if it is user facing module - rename it to `msd_inbound_policy` the same for outbound

File: example/main.tf [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+
+provider "athenz" {
+  zms_url = "https://dev.zms.athens.yahoo.com:4443/zms/v1"
+  #  zms_url = "https://stage.zms.athenz.ouroath.com:4443/zms/v1"
+  cert = "/Users/${local.user_name}/.athenz/cert" # change to your location
+  key  = "/Users/${local.user_name}/.athenz/key"  # change to your location
+}
+
+module "mendi_msd_policies" {
+  source = "../msd_module/msd_policy"
```
Comment (dvirguttman) [inline example/main.tf:30]: finally, how do use this module? without the source part

File: athenz/assertion.go (hunk 1 above)
Comment (mendi160) [reply athenz/assertion.go:182]: You will get an error during the plan stage

File: athenz/assertion.go (hunk 3 above)
Comment (mendi160) [reply athenz/assertion.go:233]: no. I just you it in line 237 un order not to repeat myself...

File: example/main.tf (hunk 6 above)
Comment (mendi160) [reply example/main.tf:30]: no, with source part (and version). the source will be the path which we decide to push the moudle

File: athenz/assertion.go (hunk 2 above)
Comment (mendi160) [reply athenz/assertion.go:221]: If you look on AssertionCondition class definition you see that the value of each key in the assertion conditionsMap is single AssertionConditionData.

**Review by dvirguttman — APPROVED** [review]

**Pull request merged**